packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -176,6 +166,12 @@ class TimeInput extends React.Component {
         meridiem = this.anteMeridiem;
       }
     }
+    if (this.props.variant === TimeUtil.FORMAT_12_HOUR && TimeUtil.getVariantFromLocale(props) === TimeUtil.FORMAT_24_HOUR) {
+      if (process.env !== 'production') {
+        // eslint-disable-next-line no-console
+        console.warn('This locale only uses 24 hour clock. The ante meridiem and post meridiem will not be displayed');
+      }
```

Comment:
we will need to reset these below variables here. to avoid carry forwarding old meridiem values.

this.anteMeridiem = '';	 
this.postMeridiem = '';

Reply:
Done